scripts/check-update.sh

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/env bash
+set -eio pipefail
+
+usage() {
+    echo "Usage:"
+    echo "  $0 AMI_TYPE"
+    echo "Example:"
+    echo "  $0 al2"
+    echo "AMI_TYPE Must be one of: al1, al2, al2023"
+}
+
+error() {
+    local msg="$1"
+    echo "ERROR: $msg"
+    usage
+    exit 1
+}
+
+readonly ami_type="$1"
+if [ -z "$ami_type" ]; then
+    error "AMI_TYPE must be provided"
+fi
+
+case "$ami_type" in
+"al1")
+    source_ami_line_nums=(2 3 4 5)
+    source_ami_line_contents=()
+    al1_file_path="./release-al1.auto.pkrvars.hcl"
+    for line_number in "${source_ami_line_nums[@]}"; do
+        previous_line_content=$(sed -n "${line_number}p" "$al1_file_path")
+        source_ami_line_contents+=("$previous_line_content")
+    done
+
+    ./generate-release-vars.sh al1
+
+    for ((i = 0; i < "${#source_ami_line_nums[@]}"; i++)); do
+        line_number=${source_ami_line_nums[$i]}
+        previous_line_content=${source_ami_line_contents[$i]}
+        updated_line_content=$(sed -n "${line_number}p" "$al1_file_path")
+        if [ "$updated_line_content" != "$previous_line_content" ]; then
+            UpdateAl1="true"
+            break
+        fi
+    done
```

Comment:
So line 1 is guaranteed to be changed every time. Sadly, that's why we have to check line by line. We also don't want to keep that line intact in the generate-release-vars because that will make it so multiple different scripts are changing the same file, which can be harder to control when testing.

Reply:
We can also save the old contents to a temporary file and do something like 
```
diff <(grep -v ami_version release.old) <(grep -v ami_version release.new)
```

Reply:
I see. Let me try this out real quick and try to add it in.

Reply:
Thanks for the elegant solution amogh, its now implemented instead of the line by line solution if you'd like to check it out.

Reply:
💯